torchrl/modules/planners/cem.py

```diff
@@ -51,13 +51,10 @@ class CEMPlanner(MPCPlannerBase):
         >>> class MyMBEnv(ModelBasedEnvBase):
         ...     def __init__(self, world_model, device="cpu", dtype=None, batch_size=None):
         ...         super().__init__(world_model, device=device, dtype=dtype, batch_size=batch_size)
-        ...         self.observation_spec = CompositeSpec(
+        ...         self.state_spec = CompositeSpec(
```

Comment:
What I had in mind is
```suggestion
        ...         self.state_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.observation_spec = CompositeSpec(
        ...             next_hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
```

and here's the full code

```python
        >>> class MyMBEnv(ModelBasedEnvBase):
        ...     def __init__(self, world_model, device="cpu", dtype=None, batch_size=None):
        ...         super().__init__(world_model, device=device, dtype=dtype, batch_size=batch_size)
        ...         self.state_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.observation_spec = CompositeSpec(
        ...             hidden_observation=UnboundedContinuousTensorSpec((4,))
        ...         )
        ...         self.action_spec = UnboundedContinuousTensorSpec((1,))
        ...         self.reward_spec = UnboundedContinuousTensorSpec((1,))
        ...
        ...     def _reset(self, tensordict: TensorDict) -> TensorDict:
        ...         tensordict = TensorDict(
        ...             {},
        ...             batch_size=self.batch_size,
        ...             device=self.device,
        ...         )
        ...         tensordict = tensordict.update(
        ...             self.state_spec.rand())
        ...         tensordict = tensordict.update(
        ...             self.observation_spec.rand())
        ...         return tensordict
```

Not sure if the observation should be named differently from the input, IDK where that comes from.

I think that with this implementation, the input at `t+1` will match the output from `t` which satisfies the autoregressive property of the planner

Reply:
I see. I will modify according to your ideas and ensure that there are no errors.